benchmark_online.py

```diff
@@ -0,0 +1,175 @@
+		load_batch = lambda x: x.numpy()
+		args.sample_rate = 8000
+	else:
+		assert args.checkpoint is not None and os.path.isfile(args.checkpoint)
+		checkpoint = torch.load(args.checkpoint, map_location='cpu')
+		args.model, args.lang, args.sample_rate, args.window_size, args.window_stride, args.window, args.num_input_features = (
+			checkpoint['args'].get(key) for key in
+			['model', 'lang', 'sample_rate', 'window_size', 'window_stride', 'window', 'num_input_features']
+		)
+
+		text_config = json.load(open(args.text_config))
+		text_pipelines = []
+		for pipeline_name in args.text_pipelines:
+			text_pipelines.append(text_processing.ProcessingPipeline.make(text_config, pipeline_name))
+
+		frontend = models.LogFilterBankFrontend(
+			args.num_input_features,
+			args.sample_rate,
+			args.window_size,
+			args.window_stride,
+			args.window,
+			stft_mode=args.stft_mode
+		)
+		model = getattr(models, args.model)(
+			num_input_features=args.num_input_features,
+			num_classes=[pipeline.tokenizer.vocab_size for pipeline in text_pipelines],
+			frontend=frontend,
+			dict=lambda logits, log_probs, olen, **kwargs: logits[0]
+		)
+		model.load_state_dict(checkpoint['model_state_dict'], strict=False)
+		model.to(args.device)
+		model.eval()
+		model.fuse_conv_bn_eval()
+		if use_cuda:
+			model, *_ = models.data_parallel_and_autocast(model, opt_level=args.fp16, data_parallel=False)
+		load_batch = lambda x: x.to(args.device, non_blocking=True)
+
+	batch_width = int(math.ceil(args.T * args.sample_rate / 128) * 128)
+	example_time = batch_width / args.sample_rate
+	batch = torch.rand(args.B, batch_width)
+	batch = batch.pin_memory()
+
+	print(f'batch [{args.B}, {batch_width}] | audio {args.B * example_time:.2f} sec\n')
+
+	def tictoc():
+		if use_cuda:
+			torch.cuda.synchronize()
+		return time.time()
+
+	print(f'Warming up for {args.warmup_iterations} iterations...')
+	tic_wall = tictoc()
+	if use_cuda:
+		torch.backends.cudnn.benchmark = True
+	torch.set_grad_enabled(False)  # no back-prop - no gradients
+	for i in range(args.warmup_iterations):
+		model(load_batch(batch))
+	print(f'Warmup done in {tictoc() - tic_wall:.1f} sec\n')
+
+	n_requests = int(round(args.benchmark_duration * args.rps))
+	print(f'Starting {args.benchmark_duration} second benchmark ({n_requests} requests, rps {args.rps:.1f})...')
+	schedule = np.random.random(n_requests) * args.benchmark_duration  # uniform random distribution of requests
+	schedule = np.sort(schedule) + tictoc()
+	print(f'avg gap between requests: {np.diff(schedule).mean() * 1e3:.1f} ms')
+	latency_times, idle_times = [], []
+	slow_warning = False
+	for t_request in schedule:
+		tic = tictoc()
+		if tic < t_request:  # no requests yet. at this point prod would wait for the next request
+			sleep_time = t_request - tic
+			idle_times.append(sleep_time)
+			time.sleep(sleep_time)
+		elif tic > t_request + 0.5 and not slow_warning:
```

Comment:
what's the meaning of 0.5 constant here? please move it to ArgumentParser with some meaningful name

Reply:
Done. Added `--max-latency` parameter